hove-io/navitia
# kraken: Add new realtime trip into new mvj and vj

**Remains to be done** :

- [x] Add physical mode into new vj + test, **JIRA** : https://jira.kisio.org/browse/NAVP-1065
- [x] Add company into new vj + test, **DONE**

**Tests cover**
- Add new meta VJ mechanism
- Add new VJ mechanism
- Activation flags mechanism (is_realtime_add_enabled, is_realtime_add_trip_enable)
- Check impacts
- Check new journey
- Tests when you send the same GTFS-RT twice
- Test adding 2 different GTFS-RT
- Reject GTFS-RT if company id doesn't match
- Reject GTFS-RT if physical mode id doesn't match

**JIRA** 
- https://jira.kisio.org/browse/NAVP-1066
- https://jira.kisio.org/browse/NAVP-1067 (added part)

**Warning**
- now, if you can't match the _company id_ in GTFS-RT with the company inside Kraken data, there is just a log to inform you (Add inside prometheus probe is better I think). In fact, in this case, this is very dangerous to acces to the company field of the VJ _new trip_ (Example: vj->company->uri) because it is a NULL ptr... It is memory access violation behavior
**Solution** In this case, our choice is to _reject the GTFS-RT new trip_. **DONE** see test
Same technic is used for physical mode id. **DONE** see test
